src/Raven.Server/Web/RequestHandler.cs

```diff
@@ -171,12 +171,19 @@ protected static GZipStream GetGzipStream(Stream stream, CompressionMode mode, C
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         protected bool ClientAcceptsGzipResponse()
         {
-
             return
                 Server.Configuration.Http.UseResponseCompression &&
                 (HttpContext.Request.IsHttps == false ||
                     (HttpContext.Request.IsHttps && Server.Configuration.Http.AllowResponseCompressionOverHttps)) &&
-                GetHttpCompressionAlgorithmFromHeaders(HttpContext.Request.Headers, Constants.Headers.AcceptEncoding) == HttpCompressionAlgorithm.Gzip;
+                HasHttpAcceptEncoding(HttpContext.Request.Headers, Constants.Headers.Encodings.Gzip);
+        }
+
+        private static bool HasHttpAcceptEncoding(IDictionary<string, Microsoft.Extensions.Primitives.StringValues> headers, string encoding)
+        {
+            if (headers.TryGetValue(Constants.Headers.AcceptEncoding, out Microsoft.Extensions.Primitives.StringValues acceptEncodings) == false)
+                return false;
+
+            return acceptEncodings.ToString().Contains(encoding);
```

Comment:
Is there a way to _not_ materialize as a string here? Can check the contents one a time, no?

Reply:
@kalczur change this to:

```
        private static bool HasHttpCompressionAlgorithmInHeaders(IDictionary<string, Microsoft.Extensions.Primitives.StringValues> headers, string encodingsHeader, string acceptedEncoding)
        {
            if (headers.TryGetValue(encodingsHeader, out Microsoft.Extensions.Primitives.StringValues acceptedContentEncodings) == false)
                return false;

            foreach (var encoding in acceptedContentEncodings)
            {
                if (encoding == acceptedEncoding)
                    return true;
            }

            return false;
        }
```

Reply:
discussed with @ppekrol